enhancements/etcd/automated-backups-no-config.md

```diff
@@ -0,0 +1,299 @@
+
+
+#### DaemonSet manages an Etcd-backup pod within every control-plane node.
+
+This approach has been initiated during reviews of the `Sidecar` method. A reviewer has recommended to separate the backup container from the static pod to mitigate the issues shown above.
+Initially a `Deployment` resource was suggested, before using a `DaemonSet`. The advantage of using DaemonSet over Deployment is to guarantee scheduling `Etcd-backup` pods on all `Control-plane` nodes, regardless of the number of nodes (e.g. 1, 3, 5) using `nodeSelector` capability.
+By setting the `nodeSelector` to the master node label (i.e. `node-role.kubernetes.io/master`), guarantees that an Etcd-backup pod will be scheduled on every control-plane node.
+
+**Pros**
+
+This method has been implemented and tested. The same container used in the Sidecar approach has been utilised. However, the scheduling is being managed by a DaemonSet resource.
+Upon creating a `Backup` CR on an Openshift cluster, the `PeriodicBackupController` reacts by creating a DaemonSet resource, with the `etcd-backup-server` container and the configurations applied in the CR.
+During [testing](https://github.com/openshift/cluster-etcd-operator/pull/1354#issuecomment-2408661261), this method has shown a solid and stable deployment, where the Backup pods shown only `1` restart over a duration of `45` minutes. Moreover, the cause of the restart on all the Backup pods were due to `node` issue, not the backup container.
+
+**Cons**
+
+No issue has been encountered during testing this approach.
+
+### Open Questions
+
+### Implementation Details/Notes/Constraints
+
+The _no-config_ backups builds atop of the existing [automated backup of etcd](https://docs.openshift.com/container-platform/4.15/backup_and_restore/control_plane_backup_and_restore/backing-up-etcd.html#creating-automated-etcd-backups_backup-etcd).
+
+The `PeriodicBackupController` within **cluster-etcd-operator** checks for `Backup` CR with `name=default` within every sync. Upon applying a `Backup` CR with `name=default`, the _no-config_ backup sidecar container is being enabled within each etcd pod in Openshift cluster.
+The `Backup` configurations such as `scheudle`, `timezone` and `retention policy`, are being supplied to the _no-config_ container as flags. 
+According to user's preference, the default configuration could be modified by updating the CR.
+
+The backups are being stored on the local disk of each master node independently, using `hostPath` volume. This replication is required for disaster recovery scenarios where more than one of the master nodes has been lost.
+
+A `Backup` CR with default configuration below.
+
+```yaml
+apiVersion: config.openshift.io/v1alpha1
+kind: Backup
+metadata:
+  name: default
+spec:
```

Comment:
How many of these fields are defaulted vs required, is something creating this default for the user?

Reply:
> How frequent are the default backups? What is the impact to the etcd cluster when the backup occurs, does it freeze the DB for any period?

So the default values are not being decided yet, and  I added this to the OpenQuestion section for discussion 

> What happens when the control plane is resized using CPMS, is there any risk of a period where there are no backups left because CPMS got rid of them?

Actually the backups will be gone, since they are being saved on `hostPath` which is on the node's file system 

Since CPMSO will delete the master nodes, the backup will be gone 

However, since this works orthogonal to the `CronJob` implementation, any backups that are saved on the mounted PVC will be available .. Also a continuation of this work is to use `external target` for saving backups, for `Disaster Recovery` scenarios, where all the control-plane nodes are gone